x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowStatsAction.java

```diff
@@ -85,7 +98,74 @@ protected void doExecute(
                 throw new ResourceNotFoundException("No shard follow tasks for follower indices [{}]", resources);
             }
         }
-        super.doExecute(task, request, listener);
+        super.doExecute(task, request, listener.map(this::enrichWithIndexDocumentCountLag));
+    }
+
+    private FollowStatsAction.StatsResponses enrichWithIndexDocumentCountLag(final FollowStatsAction.StatsResponses responses)
+        throws Exception {
+        // structure of mapping: follower index -> (remote cluster, leader index)
+        final Map<String, Tuple<String, String>> followerIndexToRemoteCluster = new HashMap<>();
+        responses.getStatsResponses()
+            .stream()
+            .map(FollowStatsAction.StatsResponse::status)
+            .forEach(
+                status -> followerIndexToRemoteCluster.computeIfAbsent(
+                    status.followerIndex(),
+                    ignored -> Tuple.tuple(status.getRemoteCluster(), status.leaderIndex())
+                )
+            );
+        followerIndexToRemoteCluster.forEach((followerIndex, remoteClusterAndIndex) -> {
+            final String remoteCluster = remoteClusterAndIndex.v1();
+            final String leaderIndex = remoteClusterAndIndex.v2();
+            try {
+                responses.setFollowerIndexDocumentCountLag(followerIndex, getDocumentCountLag(followerIndex, leaderIndex, remoteCluster));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        return responses;
+    }
+
+    private long getDocumentCountLag(final String followerIndex, final String leaderIndex, final String remoteCluster) throws Exception {
+
+        final CompletableFuture<Long> localIndexDocCount = new CompletableFuture<>();
+        spawnDocCountRequest(transportService::getLocalNodeConnection, followerIndex, localIndexDocCount::complete);
+
+        final CompletableFuture<Long> remoteIndexDocCount = new CompletableFuture<>();
+        spawnDocCountRequest(
+            () -> transportService.getRemoteClusterService().getConnection(remoteCluster),
+            leaderIndex,
+            remoteIndexDocCount::complete
+        );
+
+        return remoteIndexDocCount.get() - localIndexDocCount.get();
```

Comment:
It may be more useful to show both the local doc count, the remote doc count, and also their difference in the final stats output (rather than just the difference).

Also, while the difference of doc counts could in some situations insinuate the progress of the follower, it may not always be true. Imagine for example that the leader starts deleting documents. The follower may well have more documents in that case, but still be behind the leader, since the follower needs to delete more documents to catch up. In that case, as the parent issue says, it may be more useful to estimate the progress through the `max_seq_no` difference between the follower and the leader.